source/binder.cpp

```diff
@@ -13,6 +13,18 @@
 #include <binder.hpp>
 
 // Declares clang::SyntaxOnlyAction.
+#if  (LLVM_VERSION_MAJOR < 4)
+#include "clang/AST/ASTConsumer.h"
+#include "clang/AST/ASTContext.h"
+typedef clang::ASTConsumer* ASTConsumerPtr;
+#endif
+#if (LLVM_VERSION_MAJOR >= 4)
+#include "clang/AST/ASTConsumer.h"
+#include "clang/AST/ASTContext.h"
+#include "clang/Frontend/FrontendActions.h"
+typedef std::unique_ptr<clang::ASTConsumer> ASTConsumerPtr;
```

Comment:
This seems wrong to me: name of the type imply a pointer but unique pointer was used instead. Also, i do not thing using typedef here is justified since type is only named in single place.

Reply:
Ptr in the end of type implies that thas is  a pointer in a wide sense: raw, std::unique, std::shared, boost,  custom, etc.

Yes, it is in one place but it looks better than one mode ifdef.

Reply:
You missing my point here: deleting unique pointer will lead to extra call to destructor which lead to double-free-error because API return raw pointer (ie no-owning-semantic) instead of unique-pointer (ie owner of allocated object)